web/src/app.js

```diff
@@ -161,12 +155,39 @@ async function exportDiagram() {
     return (await modeler.saveXML({format: true}));
 }
 
+/**
+ * @param {FilesContent[]} files
+ */
+function setFilesContent(files) {
+    vscode.setState({
+        ...vscode.getState(),
+        files: JSON.stringify(files)
+    });
+    files.forEach((file) => {
+        switch (file.type) {
+            case 'config': {
+                configs = file.content;
```

Comment:
Last on wins. Wenn in den files mehrmals derselbe type enthalten ist, dann überschreiben die file.contents sich gegenseitig.
Wahrscheinlich musst du die einzelnen Elemente von file.content in das config array rein pushen